### DISCUSSION TITLE
Understanding the directories under the src directory

### DISCUSSION THREAD

**Discussion by TheSuperEngineer** [checkstyle/checkstyle#12882] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/90137881/226932796-8580d50b-de2c-4d18-b9e8-80aadfbb2525.png)

As you can see, there are few folders under the src folder. I know that the ``main`` folder contains all the main source files that do the checking etc.

The ``test`` directory contains all the test files and resources related to that. 

Then comes this ``xdocs`` folder. 

#### Is the purpose of this xdocs folder only to generate the documentation (website) of checkstyle. 
(gpt's response)
![image](https://user-images.githubusercontent.com/90137881/226933858-92875e7a-b60d-4d07-8a62-9e9c60d2c913.png)

(i have more questions incoming hence, need to get the foundation cleared.)

**Comment by rnveach** [comment, marked as answer]
What was responded about xdocs and your assumption is correct.

**Reply by romani** [reply]
Yes, chatGTP reply is correct.

**Reply by nrmancuso** [reply]
@shubh220922 please "Mark as answer" if you are satisfied with response

**Reply by TheSuperEngineer** [reply]
@nrmancuso noted. I think I can't "mark as answer". Maintainer can do that maybe. I can only close the discussion (/ mark it as resolved).

**Reply by TheSuperEngineer** [reply]
Resolved.
